kit/Kit.cpp

```diff
@@ -825,6 +825,7 @@ class Document final : public DocumentManagerInterface
             {
                 LOG_FTL("Document [" << anonymizeUrl(_url) << "] has no more views, exiting bluntly.");
                 flushTraceEventRecordings();
+                _pngPool.stop();
```

Comment:
Hokay - all of the call sites you touch look like a real opportunity for code sharing & cleanup here. I expect we need to flushTraceEventRecordings() on an EX_OK and not otherwise, we need to stop the pngPool and do this Util::forcedExit etc.   - having said that I expect the Kit to be hard killed mostly by coolwsd: are we changing that ?  it is a -far- faster to go down than trying to touch all the memory we used and was paged out to fiddle with it and so on. We could of course send a USR2 signal or something to get it to flush its logs and hard exit or somesuch but ... hmm :-)

Reply:
We do stop/exit the Kits from WSD. The thread stopping is to make sure the thread_local buffer is destroyed and flushed. It's not really necessary, except that with the check I added to track the number of thread_local buffers, it's cleaner to stop the threads (at least in the graceful case).

A misbehaving Kit will get 'kill -9' from WSD. The above is for the 'nice' case.

Reply:
Refactored.